### PR TITLE
started code refactoring

### PR DESCRIPTION
**Pull request by ohamto**
addressing #36

### CODE REVIEW

**Comment by ohamto** [comment]
jan 11 2018 Notes: 
In mbdecomp.py Molecule, Fragment, Atom classes along with a method that creates an array of fragment combinations for many-body decomposition. Files in scripts directory changed for faster testing.

Next: 
mbdecomp.py should return the energy of a fragment with a get_energy method from calculator.py
calculator.py and driver.py to be added next

File: qm_mb_energy_calculator/src/calculator.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    memory = args.memory
+    method = args.method
+    basis = args.basis
+    charge = args.charge
+    spin = args.spin
+    convergence = args.convergence
+    cycles = args.cycles
+    threshold = args.threshold
+
+    # Create a fragment based on combinations list
```
Comment (ohamto) [inline qm_mb_energy_calculator/src/calculator.py]: should be outside of get_psi4_energy()

**Comment by ohamto** [comment]
Jan 24 Notes: The refactor so far
1. driver
Iterates over configs and adds fragment energies to configs
No restart feature
Accepts coordinates
*Needs more documentation

2. mbdecomp
Has the molecule class
*energies should be stored here  
Accepts an nmer molecule consisting of n fragments
Returns nmer energies
Has a "get_energy() method" [set_energy()] agnostic of the model used
*any reference to calculate is very generic
*Needs  more documentation

3. calculator
returns energy
provides a get_energy() [calculate()]
*calculate is too generic
only contains psi4_get_energy() function
*fragment combinations outside of of the psi4_get_energy() function
*Needs more documentation

Also: test needed

**Comment by xiexr151e** [comment]
Goals for next week:

- Clean up driver.py
- Need to store energies of fragments in molecule.py
- Documentation in all files

**Comment by ohamto** [comment]
06Feb2018 notes:
argparse for commandline settings
mbdecomposition: need a way to keep track of fragments/energies
    store fragments/energies in lists

in mdecomp.py:
    mChoosen function needs to be rename e.g. build_list_of_fragment
    separate "mChoosen" from iteration

**Comment by agoetz** [comment]
Closing this pull request as it is outdated. Changes have been incorporated into newer pull requests.

**Pull request closed without merging**